### PR TITLE
Added upgrade & maintenance guidance for plugins

### PR DESCRIPTION
**Pull request by cwarnermm**
Documented plugin guidance based on cross-functional [discussion](https://hub.mattermost.com/private-core/pl/91b7ggcuc3rotpkcoz9e5ak68y).

### CODE REVIEW

File: source/guides/integrate-and-extend.rst [hunk 1]
```diff
@@ -36,3 +36,15 @@ Your Mattermost deployment comes with the following integrations you can configu
...
+
+Mattermost Marketplace integrations
+-----------------------------------
+
+Additional Mattermost integrations are available on the `Mattermost Marketplace <https://mattermost.com/marketplace/>`_. You can install these integrations directly from the Marketplace, by uploading them in the System Console, or by using the REST API.
+
+.. important::
+
+  - Manually placing Mattermost integration files in the ``/plugins`` directory isn't recommended.
```
Comment (marianunez) [inline source/guides/integrate-and-extend.rst:47]: Maybe we can include that this is planned to be deprecated in future release. We can link @lieut-data forum post announcement https://forum.mattermost.com/t/deprecation-notice-manual-plugin-deployment/21192

File: source/integrate/github-interoperability.rst [hunk 2]
```diff
@@ -102,6 +102,11 @@ Upgrade
 
 We recommend updating this integration as new versions are released. Generally, updates are seamless and don't interrupt the user experience in Mattermost. Visit the `Releases page <https://github.com/mattermost/mattermost-plugin-github/releases>`__ for the latest release, available releases, and compatibiilty considerations.
 
+.. important::
+
+  - We strongly recommend removing older versions of this integration before uploading a new version. Mattermost server selects the newest integration version automatically, unless :ref:`automatic prepackaged plugins <configure/plugins-configuration-settings:automatic prepackaged plugins>` is disabled because precise control over plugin versions is preferred.
+  - Use the System Console, REST API, or in-product Marketplace to manage this integration. Manually placing Mattermost integration files in the ``/plugins`` directory isn't recommended.
```
Comment (marianunez) [inline source/integrate/github-interoperability.rst:108]:
```suggestion
  - Use the System Console, REST API, or in-product Marketplace to manage this integration. Manually placing Mattermost integration files in the ``/plugins`` directory isn't recommended and will be [deprecated in future versions](https://forum.mattermost.com/t/deprecation-notice-manual-plugin-deployment/21192).
```

**Review by marianunez — APPROVED** [review]

File: source/guides/integrate-and-extend.rst (hunk 1 above)
Comment (cwarnermm) [inline source/guides/integrate-and-extend.rst:47]:
```suggestion
  - Manually placing Mattermost integration files in the ``/plugins`` directory isn't recommended and will be `deprecated in future versions <https://forum.mattermost.com/t/deprecation-notice-manual-plugin-deployment/21192>`_.
```

File: source/guides/integrate-and-extend.rst [hunk 3]
```diff
@@ -36,3 +36,15 @@ Your Mattermost deployment comes with the following integrations you can configu
...
+-----------------------------------
+
+Additional Mattermost integrations are available on the `Mattermost Marketplace <https://mattermost.com/marketplace/>`_. You can install these integrations directly from the Marketplace, by uploading them in the System Console, or by using the REST API.
+
+.. important::
+
+  - Manually placing Mattermost integration files in the ``/plugins`` directory isn't recommended and will be `deprecated in future versions <https://forum.mattermost.com/t/deprecation-notice-manual-plugin-deployment/21192>`_.
+  - Mattermost uses the latest integration version available within the following directories: ``<installdirectory>/prepackaged-plugins`` and ``FileSettings.Directory/plugins``, where the latter is the source of truth for installed plugins and contains ``*.tar.gz`` files.
+  - We strongly recommend that system admins ensure that ``{{FileStore.Directory}}/plugins`` is mounted at startup to avoid integration downgrades.
+  - We also strongly recommend testing integration updates in a staging environment before deploying to production, and regularly backing up integrations.
```
Comment (lieut-data) [inline source/guides/integrate-and-extend.rst:50]: Propose a simpler summary for all of these:
> Installed plugins are persisted to the configured file store and unpacked on server startup. It's imperative that your file store be accessible to the server immediately on startup. If using a shared filesystem, ensure the mount completes successfully before starting the server.

File: source/integrate/github-interoperability.rst [hunk 4]
```diff
@@ -102,6 +102,11 @@ Upgrade
 
 We recommend updating this integration as new versions are released. Generally, updates are seamless and don't interrupt the user experience in Mattermost. Visit the `Releases page <https://github.com/mattermost/mattermost-plugin-github/releases>`__ for the latest release, available releases, and compatibiilty considerations.
 
+.. important::
+
+  - We strongly recommend removing older versions of this integration before uploading a new version. Mattermost server selects the newest integration version automatically, unless :ref:`automatic prepackaged plugins <configure/plugins-configuration-settings:automatic prepackaged plugins>` is disabled because precise control over plugin versions is preferred.
+  - Use the System Console, REST API, or in-product Marketplace to manage this integration. Manually placing Mattermost integration files in the ``/plugins`` directory isn't recommended and will be `deprecated in future versions <https://forum.mattermost.com/t/deprecation-notice-manual-plugin-deployment/21192>`_.
+
```
Comment (lieut-data) [inline source/integrate/github-interoperability.rst:109]: From here on out, I'd propose reverting all the changes. We don't need to duplicate this on a per-plugin basis, IMO.

**Review by lieut-data — CHANGES_REQUESTED** [review]
Thanks, Carrie -- a few thoughts below.

**Comment by cwarnermm** [comment]
Thanks, @lieut-data! I've incorporated your feedback in my latest commit.

**Pull request merged**